### PR TITLE
AURO MIGRATION: Update node to v22

### LINKED ISSUE #6: Update defined node versions for latest security requirements
### Please describe the issue

Component templates need to be using Node 20 and 22.

### Exit criteria

All references to node 18 and 20 are updated to 20 and 22.

### PR DESCRIPTION
**Pull request by jason-capsule42**
Resolves AlaskaAirlines/auro-templates#6

## Summary by Sourcery

Update Node.js to version 22.

Build:
- Update Node engine to support versions 20 and 22.

CI:
- Update CI workflow to test on Node versions 20 and 22.

Documentation:
- Update contributing guidelines with information about rebasing, conventional commits, and issue/PR templates.

Tests:
- Update test matrix to include Node version 22.

### CODE REVIEW

**Review by sun-mota — CHANGES_REQUESTED** [review]
Need https://github.com/AlaskaAirlines/auro-menu/pull/233

**Review by sun-mota — APPROVED** [review]

File: .github/workflows/testPublish.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
           persist-credentials: false
       - uses: actions/setup-node@v4
         with:
-          node-version: 20.x
+node-version: 22.x
```
Comment (sun-mota) [inline .github/workflows/testPublish.yml]: wrong indent

**Review by sun-mota — CHANGES_REQUESTED** [review]

File: .github/workflows/testPublish.yml (hunk 1 above)
Comment (rmenner) [reply .github/workflows/testPublish.yml]: That would break it for sure

Comment (DukeFerdinand) [reply .github/workflows/testPublish.yml]: fixed!

**Review by sun-mota — APPROVED** [review]

**Pull request merged**